types/websocket/index.d.ts

```diff
@@ -564,10 +565,10 @@ declare class client extends events.EventEmitter {
      *                 any scripting content that caused the connection to be requested.
      * @param requestUrl should be a standard websocket url
      */
-    connect(requestUrl: url.Url, protocols?: string[], origin?: string, headers?: any[], extraRequestOptions?: http.RequestOptions): void;
-    connect(requestUrl: string, protocols?: string[], origin?: string, headers?: any[], extraRequestOptions?: http.RequestOptions): void;
-    connect(requestUrl: url.Url, protocols?: string, origin?: string, headers?: any[], extraRequestOptions?: http.RequestOptions): void;
-    connect(requestUrl: string, protocols?: string, origin?: string, headers?: any[], extraRequestOptions?: http.RequestOptions): void;
+    connect(requestUrl: url.Url, protocols?: string[], origin?: string, headers?: {}, extraRequestOptions?: http.RequestOptions): void;
```

Comment:
`{}` will allow `number` and `string` (since it's structural). Perhaps you'd prefer `object` instead, which will only allow structured types and not primitives.